frontend/test/metabase/scenarios/admin/subscription/payment-failure.cy.spec.js

```diff
@@ -0,0 +1,52 @@
+import { restore, mockSessionProperty } from "__support__/e2e/helpers";
+
+describe("banner", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("Show a banner when the subscription payment status is `past-due`", () => {
+    mockSessionProperty("token-status", {
+      status: "past-due",
+      valid: false,
+      trial: false,
+      features: [],
+    });
+
+    cy.visit("/");
+    cy.contains("We couldn't process payment for your account.");
+    cy.visit(`/admin/`);
+    cy.contains("We couldn't process payment for your account.");
+
+    cy.signInAsNormalUser();
+    cy.visit("/");
+    // Wait for page to load
+    cy.get("header");
+    cy.contains("We couldn't process payment for your account.").should(
+      "not.exist",
+    );
+  });
+
+  it("Show a banner when the subscription payment status is `unpaid`", () => {
+    mockSessionProperty("token-status", {
+      status: "unpaid",
+      valid: false,
+      trial: false,
+      features: [],
+    });
+
+    cy.visit("/");
+    cy.contains("Pro features won’t work right now due to lack of payment.");
```

Comment:
We use testing-library methods, e.g. here it would be `cy.findByText("...")`
And we also explicitly add assertions, e.g. `cy.findByText("...").should("be.visible")`